x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/index.test.tsx

```diff
@@ -1391,4 +1391,202 @@ describe('when on the endpoint list page', () => {
+      )[tableRow];
+
+      reactTestingLibrary.act(() => {
+        reactTestingLibrary.fireEvent.click(endpointActionsButton);
+      });
+    };
+
+    beforeEach(async () => {
+      const { data: hosts } = mockEndpointResultList({ total: 2 });
+      // second host is isolated, for unisolate testing
+      const hostInfo: HostInfo[] = [
+        {
+          host_status: hosts[0].host_status,
+          metadata: {
+            ...hosts[0].metadata,
+            Endpoint: {
+              ...hosts[0].metadata.Endpoint,
+              capabilities: [...ENDPOINT_CAPABILITIES],
+              state: {
+                ...hosts[0].metadata.Endpoint.state,
+                isolation: false,
+              },
+            },
+            host: {
+              ...hosts[0].metadata.host,
+              os: {
+                ...hosts[0].metadata.host.os,
+                name: 'Windows',
+              },
+            },
+            agent: {
+              ...hosts[0].metadata.agent,
+              version: '7.14.0',
+            },
+          },
+        },
+        {
+          host_status: hosts[1].host_status,
+          metadata: {
+            ...hosts[1].metadata,
+            Endpoint: {
+              ...hosts[1].metadata.Endpoint,
+              capabilities: ['isolation'],
+              state: {
+                ...hosts[1].metadata.Endpoint.state,
+                isolation: true,
+              },
+            },
+            host: {
+              ...hosts[1].metadata.host,
+              os: {
+                ...hosts[1].metadata.host.os,
+                name: 'Windows',
+              },
+            },
+            agent: {
+              ...hosts[1].metadata.agent,
+              version: '8.4.0',
+            },
+          },
+        },
+      ];
+      setEndpointListApiMockImplementation(coreStart.http, {
+        endpointsResults: hostInfo,
+        endpointPackagePolicies: mockPolicyResultList({ total: 2 }).items,
+      });
+    });
+    afterEach(() => {
+      jest.clearAllMocks();
+      mockUserPrivileges.mockReset();
+    });
+    it('shows Isolate host option if canHostIsolate is READ/ALL', async () => {
+      mockUserPrivileges.mockReturnValue({
+        ...mockInitialUserPrivilegesState(),
+        endpointPrivileges: {
+          ...mockInitialUserPrivilegesState().endpointPrivileges,
+          canIsolateHost: true,
+        },
+      });
+      renderAndClickActionsButton();
+      const isolateLink = await renderResult.findByTestId('isolateLink');
+      expect(isolateLink).not.toBeNull();
+    });
+    it('hides Isolate host option if canIsolateHost is NONE', () => {
+      mockUserPrivileges.mockReturnValue({
+        ...mockInitialUserPrivilegesState(),
+        endpointPrivileges: {
+          ...mockInitialUserPrivilegesState().endpointPrivileges,
+          canIsolateHost: false,
+        },
+      });
+      renderAndClickActionsButton();
```

Comment:
We need the `await` keyword before every `renderAndClickActionsButton()` function call, because tests using `screen.queryByTestId()` now pass accidentally, as they don't wait for the rendering to be completed. It could happen that the queried element does not exist _yet_, but will exist later.

Reply:
hi so I tried using await in front of that function like so, `await renderAndClickActionsButton()` but I get this kind of error which I don't think is supposed to happen 
<img width="1722" alt="image" src="https://user-images.githubusercontent.com/56409205/205344449-fa947c0d-7f42-49a1-b0bf-4dd192ef6c65.png">

Reply:
I think the `async` mark may be missing - you need to mark every function as `async` if you use `await` inside them, like this:
```TypeScript
it('...', async () => {
   await doStuff();
}
```